File: card-lang-rust/src/calculator/parser/symbols/number.rs
use std::collections::HashMap;

use crate::types::Parser::Parser;
use crate::types::Parser::NudListener;
use crate::types::Parser::SymbolAndToken;
use crate::types::ReusedStructs::Node;
use crate::types::ReusedStructs::NodeType;

pub struct NumberNud {}

impl NudListener for NumberNud {
  fn run(&self, symtok: SymbolAndToken, _: Parser)->Node{
    return Node {
      node_type: NodeType::ValueNode,
      values: Some(HashMap::from([
        ("isFLoat".to_string(), symtok.token.values.get("isFloat").unwrap().clone()),
        ("value".to_string(), symtok.token.values.get("value").unwrap().clone()),
      ])),
      branches: None,
      args: None
    };
  }
}

File: card-lang-typescript/src/calculator/parser/symbols/number.ts
import { NudListener, ParserController, SymbolAndToken } from "../../../types/Parser";
import { LangNode, LangNodeType } from "../../../types/ReusedStructs";

export class NumberNud implements NudListener {
  run(symtok: SymbolAndToken, _: ParserController): LangNode {
    return {
      node_type: LangNodeType.ValueNode,
      values: {
        type: "number",
        is_float: symtok.token.values.is_float == "1",
        value: symtok.token.values.value
      }
    };
  }
}
